internal/services/oracledatabase/exadata_infrastructure_resource.go

```diff
@@ -0,0 +1,337 @@
+// Copyright © 2024, Oracle and/or its affiliates. All rights reserved
+
+package oracledatabase
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/zones"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2024-06-01/cloudexadatainfrastructures"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"time"
+)
+
+var _ sdk.Resource = ExadataInfraResource{}
+
+type ExadataInfraResource struct{}
+
+type ExadataInfraResourceModel struct {
+	// Azure
+	Location          string                 `tfschema:"location"`
+	Name              string                 `tfschema:"name"`
+	ResourceGroupName string                 `tfschema:"resource_group_name"`
+	Tags              map[string]interface{} `tfschema:"tags"`
+	Zones             zones.Schema           `tfschema:"zones"`
+
+	// Required
+	ComputeCount int64  `tfschema:"compute_count"`
+	DisplayName  string `tfschema:"display_name"`
+	Shape        string `tfschema:"shape"`
+	StorageCount int64  `tfschema:"storage_count"`
+
+	// Optional
+	CustomerContacts  []string                 `tfschema:"customer_contacts"`
+	MaintenanceWindow []MaintenanceWindowModel `tfschema:"maintenance_window"`
+}
+
+func (ExadataInfraResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		// Azure
+		"location": commonschema.Location(),
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"resource_group_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"tags":  commonschema.Tags(),
+		"zones": commonschema.ZonesMultipleRequired(),
```

Comment:
these should be at the bottom
```suggestion
		"tags":  commonschema.Tags(),
		
		"zones": commonschema.ZonesMultipleRequired(),
```

Reply:
Done thanks!